### PR TITLE
uBPF behavior around division and modulus is non-conformant

### PR DESCRIPTION
**Pull request by Alan-Jowett**
Linux behavior for EBPF_OP_DIV* and EBPF_OP_MOD* are different from the way uBPF handles them.

uBPF behavior:
Division by zero -> Terminate the program and return -1.
Modules by zero -> Terminate the program and return -1.

Specifically:
Division by zero -> Don't terminate the program and set the result to 0.
Modulus by zero -> Don't terminate the program and set the result to the dividend. 

This PR changes the behavior to match Linux.

Signed-off-by: Alan Jowett <alanjo@microsoft.com>

Resolves: #115

### CODE REVIEW

**Comment by Alan-Jowett** [comment]
@matt-gretton-dann Could you please make the changes to the ARM64 case? I am not sufficiently familiar with the ARM64 architecture to confidently do it.

**Comment by Alan-Jowett** [comment]
@qmonnet Quentin, you might want to update [rBPF](https://github.com/qmonnet/rbpf). Not sure the approach I took is best, but it does avoid having to add support for jumping to a non-BPF instruction boundary.

**Comment by matt-gretton-dann** [comment]
@Alan-Jowett : Slight change in company role for me so I need to pass this on.  I'll ensure someone looks at it.

File: tests/err-div-by-zero-imm.data [hunk 1]
```diff
@@ -2,5 +2,5 @@
 mov32 r0, 1
 div32 r0, 0
 exit
--- error
-Failed to load code: division by zero at PC 1
+-- result
+0x0
```
Comment (dthaler) [inline tests/err-div-by-zero-imm.data:6]: there seems to be a missing test case for err-mod-by-zero-imm.data which doesn't exist.  Can you create one?

File: vm/ubpf_jit_x86_64.c [hunk 2]
```diff
@@ -515,42 +515,74 @@ translate(struct ubpf_vm *vm, struct jit_state *state, char **errmsg)
...
+
+        // Save the dividend for the modulo case.
+        if (mod) {
+            emit_push(state, RAX);  // Save dividend.
+        }
+
+        // Save the result of the test.
+        emit1(state, 0x9c); /* pushfq */
+
+        // Set the divisor to 1 of it is zero.
```
Comment (dthaler) [inline vm/ubpf_jit_x86_64.c:579]:
```suggestion
        // Set the divisor to 1 if it is zero.
```

File: tests/err-div-by-zero-imm.data (hunk 1 above)
Comment (Alan-Jowett) [reply tests/err-div-by-zero-imm.data]: Good catch. There was actually a bug here.
